docs/usage/configuration/config_documentation.md

```diff
@@ -3765,6 +3806,109 @@ redis:
   port: 6379
   password: <secret_password>
 ```
+---
+## Worker configuration for workers configuration
+These switches configure each worker them self with the a worker configuration file.
+
+Note also the configuration for the
+[main process above](#worker-configuration-for-main-process).
+
+The manual how to configure workers is described in detail in the
+[worker documentation](../../workers.md)
+
+---
+### `worker_app`
+
+The type of worker. The currently available worker applications are listed
+in [worker documentation](../../workers.md#available-worker-applications).
+
+The most common worker is the
+[`synapse.app.generic_worker`](../../workers.md#synapseappgeneric_worker).
+
+Example configuration:
+```yaml
+worker_app: synapse.app.generic_worker
+```
+---
+### `worker_name`
+
+A unique name for the worker. The worker needs a name to be addressed in
+further parameters and identification in log files.
+
+Example configuration:
+```yaml
+worker_name: generic_worker1
+```
+---
+### `worker_replication_host`
+
+The HTTP replication endpoint that it should talk to on the main Synapse process.
+The main Synapse process defines this with a `replication` resource in
+[`listeners` option](#listeners).
+
+Example configuration:
+```yaml
+worker_replication_host: 127.0.0.1
+```
+---
+### `worker_replication_http_port`
+
+The HTTP replication port that it should talk to on the main Synapse process.
+The main Synapse process defines this with a `replication` resource in
+[`listeners` option](#listeners).
+
+Example configuration:
+```yaml
+worker_replication_http_port: 9093
+```
+---
+### `worker_listeners`
+
+A worker can handle HTTP requests. If handling HTTP requests, a `worker_listeners`
+option with an http listener, in the same way as the [`listeners` option](#listeners)
+in the shared config.
+
+Example configuration:
+```yaml
+worker_listeners:
+  - type: http
+    port: 8083
+    resources:
+      - names: [client, federation]
+```
+---
+### `worker_daemonize`
+
+Specifies whether the worker should be daemonize. If
+[systemd](../../systemd-with-workers/README.md) is used, this must not configured.
+Systemd manages daemonization itself. Defaults to `false`.
+
+Example configuration:
+```yaml
+worker_daemonize: true
+```
+---
```

Comment:
Can you briefly explain what daemonization is (e.g. a link to wikipedia or some similar reference)?

Reply:
Honestly, no. And I didn't find anything about it in the Synapse docs.

Reply:
The short answer is if you have a process orchestrating(like `systemd` or `supervisord` the start and stop of the individual workers and master process, then you don't want to use this. If you are running, for example, on multiple machines and start the process with scripting starting each process or `synctl` then this is useful. The [Wikipedia](https://en.wikipedia.org/wiki/Daemon_(computing)) entry is exhaustive reading and not particularly useful when someone just wants to know "Do I need this?"

Reply:
Thanks both. Let's leave this as it is.